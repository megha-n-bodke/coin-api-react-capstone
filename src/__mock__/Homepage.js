import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { ListGroup } from 'react-bootstrap';

const Homepage = () => {
  const coins = [{ id: 1, name: 'Bitcoins', symbol: 'btc' }];
  return (
    <BrowserRouter>
      <h3>Currency</h3>
      <ListGroup key={coins.id}>
        {coins.map((coins) => (
          <ListGroup.Item key={coins.id}>
            <h5>{coins.name}</h5>
            <p>{coins.symbol}</p>
          </ListGroup.Item>
        ))}
      </ListGroup>
    </BrowserRouter>
  );
};

export default Homepage;
