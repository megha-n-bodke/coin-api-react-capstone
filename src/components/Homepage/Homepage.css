.col-6 {
  height: 100px;
  width: 50%;
}

.col-6:nth-child(4n + 1),
.col-6:nth-child(4n + 4) {
  background-color: #ce4177;
}

.coin-image {
  height: 30px;
  width: 30px;
}

.navigation-arrow {
  margin-left: 170px;
}
