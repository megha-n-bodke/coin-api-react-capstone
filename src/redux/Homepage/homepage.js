import { createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

const GETCOINS = 'GETCOINS';
const GETCOINSDETAILS = 'GETCOINSDETAILS';
const baseUrl = 'https://api.coingecko.com/api/v3/coins/markets?vs_currency=usd&order=market_cap_desc&per_page=50&page=1&sparkline=false';
const coindetailsUrl = 'https://api.coingecko.com/api/v3/coins';

// Action creator
export const getConis = createAsyncThunk(
  GETCOINS,
  async (args, { dispatch }) => {
    const { data } = await axios.get(baseUrl);

    dispatch({
      type: GETCOINS,
      payload: data,
    });
  },
);

// action for coin details by id
export const getCoinsDetails = createAsyncThunk(
  GETCOINSDETAILS,
  async (id, { dispatch }) => {
    const { data } = await axios.get(`${coindetailsUrl}/${id}`);

    dispatch({
      type: GETCOINSDETAILS,
      payload: data,
    });
  },
);

// reducer
const initialstate = { coins: [], companyId: {} };
const coinData = (state = initialstate, action) => {
  switch (action.type) {
    case GETCOINS:
      return {
        ...state,
        coins: action.payload,
        // coins is from initial state
      };

    case GETCOINSDETAILS: {
      console.log('this is details action');
      console.log(action.paylod);
      return {
        ...state,
        companyId: action.payload,
      };
    }
    default:
      return state;
  }
};

export default coinData;
