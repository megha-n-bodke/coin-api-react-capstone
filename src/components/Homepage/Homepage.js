import React, { useState } from 'react';
import { FaArrowCircleRight, FaSearch } from 'react-icons/fa';
import { useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import './Homepage.css';
import '../Navbar/navbar.css';

const Homepage = () => {
  const [search, setSearch] = useState('');
  const navigate = useNavigate();
  // allcoins comes from reducer alias in configure store.
  const { coins } = useSelector((state) => state.allcoins);

  return (
    <>
      <form className="form-inline ">
        <input
          type="search"
          placeholder="Search"
          aria-label="Search"
          onChange={(e) => setSearch(e.target.value)}
        />
        <FaSearch className="search-icon my-4" />
      </form>
      <div className="row ps-2">
        <>
          {coins
            .filter((coin) => {
              if (search === '') {
                return coin;
              }
              if (coin.name.toLowerCase().includes(search.toLowerCase())) {
                return coin;
              }
              return null;
            })
            .map((coin) => (
              <div
                key={coin.id}
                className="col-6 d-flex flex-column py-2 h-25 w-50 "
              >
                <FaArrowCircleRight
                  className="my-2 navigation-arrow"
                  onClick={() => navigate(`/details/${coin.id}`)}
                />
                <img src={coin.image} className="coin-image" alt="" />
                <p key={Math.random()}>{coin.name}</p>
                <p>{coin.symbol}</p>
              </div>
            ))}
        </>
      </div>
    </>
  );
};

export default Homepage;
