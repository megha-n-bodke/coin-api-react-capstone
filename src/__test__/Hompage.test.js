import renderer from "react-test-renderer";
import Homepage from "../__mock__/Homepage";

describe("Home", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Homepage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
