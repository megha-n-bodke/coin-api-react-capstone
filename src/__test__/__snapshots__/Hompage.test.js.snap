// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
Array [
  <h3>
    Currency
  </h3>,
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <div
      className="list-group-item"
      data-rr-ui-event-key={null}
      onClick={[Function]}
    >
      <h5>
        Bitcoins
      </h5>
      <p>
        btc
      </p>
    </div>
  </div>,
]
`;
